src/Uno.UWPSyncGenerator/Generator.cs

```diff
@@ -136,36 +136,45 @@ abstract class Generator
 		static Generator()
 		{
 			RegisterAssemblyLoader();
-		}
-
-		public virtual void Build(string basePath, string baseName, string sourceAssembly)
-		{
 			Directory.SetCurrentDirectory(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location));
 			InitializeRoslyn();
+		}
 
+		public virtual async Task Build(string basePath, string baseName, string sourceAssembly)
+		{
 			Console.WriteLine($"Generating for {baseName} {sourceAssembly}");
 
-			_referenceCompilation = LoadUWPReferenceProject(@"..\..\..\Uno.UWPSyncGenerator.Reference\references.txt");
+			s_referenceCompilation ??= await LoadUWPReferenceProject(@"..\..\..\Uno.UWPSyncGenerator.Reference\references.txt");
+
+			_dependencyPropertySymbol = s_referenceCompilation.GetTypeByMetadataName(BaseXamlNamespace + ".DependencyProperty");
+
+			var iOSCompilationTask = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-ios");
+			var androidCompilationTask = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-android");
+			var unitTestsCompilationTask = LoadProject($@"{basePath}\{baseName}.Tests.csproj", "net7.0");
+			var macCompilationTask = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-macos");
 
-			_dependencyPropertySymbol = _referenceCompilation.GetTypeByMetadataName(BaseXamlNamespace + ".DependencyProperty");
+			var netstdReferenceCompilationTask = LoadProject($@"{basePath}\{baseName}.Reference.csproj", "net7.0");
+			var wasmCompilationTask = LoadProject($@"{basePath}\{baseName}.Wasm.csproj", "net7.0");
+			var skiaCompilationTask = LoadProject($@"{basePath}\{baseName}.Skia.csproj", "net7.0");
 
-			_iOSCompilation = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-ios");
-			_androidCompilation = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-android");
-			_unitTestsCompilation = LoadProject($@"{basePath}\{baseName}.Tests.csproj", "net7.0");
-			_macCompilation = LoadProject($@"{basePath}\{baseName}.netcoremobile.csproj", "net7.0-macos");
+			await Task.WhenAll(iOSCompilationTask, androidCompilationTask, unitTestsCompilationTask, macCompilationTask, netstdReferenceCompilationTask, wasmCompilationTask, skiaCompilationTask);
```

Comment:
I'm thinking this would be faster.

Reply:
Well. That didn't work:

```
  Unhandled exception. System.InvalidOperationException: The operation cannot be completed because a build is already in progress.
     at Microsoft.Build.Shared.ErrorUtilities.ThrowInvalidOperation(String resourceName, Object[] args)
     at Microsoft.Build.Execution.BuildManager.BeginBuild(BuildParameters parameters)
     at Microsoft.CodeAnalysis.MSBuild.Build.ProjectBuildManager.StartBatchBuild(IDictionary`2 globalProperties)
     at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.LoadAsync(CancellationToken cancellationToken)
     at Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.LoadProjectInfoAsync(String projectFilePath, ProjectMap projectMap, IProgress`1 progress, ILogger msbuildLogger, CancellationToken cancellationToken)
     at Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.OpenProjectAsync(String projectFilePath, ILogger msbuildLogger, IProgress`1 progress, CancellationToken cancellationToken)
     at Uno.UWPSyncGenerator.Generator.InnerLoadProject(String projectFile, String targetFramework) in C:\a\1\s\src\Uno.UWPSyncGenerator\Generator.cs:line 1801
     at Uno.UWPSyncGenerator.Generator.LoadProject(String projectFile, String targetFramework) in C:\a\1\s\src\Uno.UWPSyncGenerator\Generator.cs:line 1758
     at Uno.UWPSyncGenerator.Generator.Build(String basePath, String baseName, String sourceAssembly) in C:\a\1\s\src\Uno.UWPSyncGenerator\Generator.cs:line 160
     at Uno.UWPSyncGenerator.Program.Main(String[] args) in C:\a\1\s\src\Uno.UWPSyncGenerator\Program.cs:line 24
     at Uno.UWPSyncGenerator.Program.<Main>(String[] args)
```

Reply:
Yep, I tried that as well, msbuild can only work one per process. I wondered if remoting could be used, but that does not help, neither does ALC.